### PR TITLE
Remove xfail from passing tests

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Write a good description on what the PR does.
- [x] Create a news fragment with `towncrier create <IssueNumber>.<type>` which will be
  included in the changelog. `<type>` can be one of: new_check, removed_check, extension,
  false_positive, false_negative, bugfix, other, internal. If necessary you can write
  details or offer examples on how the new change is supposed to work.
- [x] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

@DudeNr33 Are these test passing correctly? Or should they be made to actually fail again?

### CODE REVIEW

File: tests/pyreverse/test_diadefs.py [hunk 1]
```diff
@@ -115,7 +114,6 @@ def test_extract_relations(self, HANDLER: DiadefsHandler, PROJECT: Project) -> N
         relations = _process_relations(cd.relationships)
         assert relations == self._should_rels
 
-    @pytest.mark.xfail
     def test_functional_relation_extraction(
         self, default_config: PyreverseConfig, get_project: GetProjectCallable
     ) -> None:
```
Comment (Pierre-Sassoulas) [inline tests/pyreverse/test_diadefs.py:119]: Line 123/124 would also needs to be updated here.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

**Review by DudeNr33 — APPROVED** [review]
The `xfail` was added after the update to `astroid` 2.12, and AFAIR we suspected the handling of implicit namespace packages to be the reason the tests were failing.
As all other tests were still passing we went with just marking these two as `xfail`.

If they are now passing again, all the better!

Pierre already mentioned the comment that should be removed again, besides that I have nothing more to add. 👍

File: tests/pyreverse/test_diadefs.py (hunk 1 above)
Comment (DanielNoord) [reply tests/pyreverse/test_diadefs.py:119]: @DudeNr33 Are we sure about that?
The comment was added 14 years ago in this commit https://github.com/PyCQA/pylint/commit/7af79e602152aa122328063d5d7d5a16c65e594a

I don't think it is related to the `xfail`?

Comment (DudeNr33) [reply tests/pyreverse/test_diadefs.py:119]: Oh, you are right. Sorry. To be honest, if the comment is this old I have no idea what "environment problem" is really meant here.

Comment (Pierre-Sassoulas) [reply tests/pyreverse/test_diadefs.py:119]: Yeah I also did not understand what the comment meant. Should we remove anyway ?

Comment (DanielNoord) [reply tests/pyreverse/test_diadefs.py:119]: I don't think so; https://en.wikipedia.org/wiki/Wikipedia:Chesterton's_fence

Comment (Pierre-Sassoulas) [reply tests/pyreverse/test_diadefs.py:119]: This Chesterton guy must have had a lot of documentation stored in his attic 😄

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by jacobtylerwalls** [comment]
I agree with removing the xfails, as I remember adding them in the astroid 2.12 upgrade.

I agree with @DanielNoord and @Chesterton about leaving the comment as is. It's too bad it's vague, but it signals a lack of confidence about something in the test result even if the assertions are okay. It might be more evident when looking at the output visually.

**Review by jacobtylerwalls — APPROVED** [review]

**Pull request merged**